op-dispute-mon/mon/withdrawals.go

```diff
@@ -0,0 +1,55 @@
+package mon
+
+import (
+	"github.com/ethereum-optimism/optimism/op-dispute-mon/mon/types"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+	"golang.org/x/exp/maps"
+)
+
+type WithdrawalMetrics interface {
+	RecordWithdrawalRequests(delayedWeth common.Address, matches bool, count int)
+}
+
+type WithdrawalMonitor struct {
+	logger  log.Logger
+	metrics WithdrawalMetrics
+}
+
+func NewWithdrawalMonitor(logger log.Logger, metrics WithdrawalMetrics) *WithdrawalMonitor {
+	return &WithdrawalMonitor{
+		logger:  logger,
+		metrics: metrics,
+	}
+}
+
+func (w *WithdrawalMonitor) CheckWithdrawals(games []*types.EnrichedGameData) {
+	matching := make(map[common.Address]int)
+	divergent := make(map[common.Address]int)
+	for _, game := range games {
+		matches, diverges := w.validateGameWithdrawals(game)
+		matching[game.WETHContract] += matches
+		divergent[game.WETHContract] += diverges
+	}
+	for contract, count := range matching {
+		w.metrics.RecordWithdrawalRequests(contract, true, count)
+	}
+	for contract, count := range divergent {
+		w.metrics.RecordWithdrawalRequests(contract, false, count)
+	}
+}
+
+func (w *WithdrawalMonitor) validateGameWithdrawals(game *types.EnrichedGameData) (int, int) {
+	recipients := maps.Keys(game.WithdrawalRequests)
+	matching := 0
+	divergent := 0
+	for _, recipient := range recipients {
+		if game.WithdrawalRequests[recipient].Amount.Cmp(game.Credits[recipient]) == 0 {
```

Comment:
nit: Can we just do?
```suggestion
	matching := 0
	divergent := 0
	for recipient, withdrawalAmount := range game.WithdrawalRequests {
		if withdrawalAmount.Cmp(game.Credits[recipient]) == 0 {
```

Though I suspect we should have a nil check for `game.Credits` in case there's a withdrawal for someone who doesn't have any credit (shouldn't happen but that's the type of bug we're trying to detect here).